spark/src/main/scala/geotrellis/spark/io/FilteringLayerReader.scala

```diff
@@ -1,17 +1,26 @@
 package geotrellis.spark.io
 
 import geotrellis.spark.Boundable
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.io.json._
+import spray.json.JsonFormat
 
-abstract class FilteringLayerReader[ID, K: Boundable, M, ReturnType] extends LayerReader[ID, ReturnType] {
+abstract class FilteringLayerReader[ID, K: Boundable, M, ReturnType] extends LayerReader[ID, K, ReturnType] {
 
-  def read(id: ID, rasterQuery: RDDQuery[K, M], numPartitions: Int): ReturnType
+  def read[I <: KeyIndex[K]: JsonFormat](id: ID, rasterQuery: RDDQuery[K, M], numPartitions: Int, format: JsonFormat[I]): ReturnType
+
+  def read[I <: KeyIndex[K]: JsonFormat](id: ID, numPartitions: Int, format: JsonFormat[I]): ReturnType =
```

Comment:
Just a way to resolve overloads @lossyrob suggested this idea (may be I loosed smth). But any comments are appreciated: there is another way of solving this issue to give special names for functions with same types.